src/iuse.cpp

```diff
@@ -871,8 +887,17 @@ int iuse::poison(player *p, item *it, bool, const tripoint& )
 {
     if ((p->has_trait("EATDEAD"))) {
         return it->type->charges_to_use();
+    }
+
+    // NPCs have a magical sense of what is inedible
+    // Players can abuse the crafting menu instead...
+    if( !it->has_flag( "HIDDEN_POISON" ) &&
+        ( p->is_npc() ||
+          !p->query_yn( _("Are you sure you want to eat this? It looks poisonous...") ) ) ) {
```

Comment:
The exploit isn't very exploity here:
If the NPC doesn't want it, it is poisonous. But if the NPC wants it, you no longer have a shroom - the NPC won't tell you "I could eat it but I'm full", it's always either just eat or refuse.

Plus, having to haul around a NPC just to smell the shrooms (that stop hiding their status at 3 survival) makes the requirements for using it a bit too steep for it to appear in any exploit guide. Shrooms don't even spawn in considerable numbers until autumn.